### ISSUE TITLE
Hot keys don't work on Mac

### ISSUE THREAD

**Issue by blaekmajik** [richrobber2/canvas-zoom#62]
Hotkeys A and T don't work on Mac (I've tried multiple browsers). Sometime T does work, but "A" - never. Checked on Safari, Chrome, Firefox, Opera

**Comment by daswer123** [comment]
https://github.com/richrobber2/canvas-zoom/issues/57
Hi, there have already been posts about this. 
I seem to have added support, but it seems that it does not work, unfortunately there is no way to properly test the functionality. If there are any results I will let you know

**Comment by ro-mak** [comment]
I had to rerun webui for it to work, but it works on m1 pro. Installed today the latest version

**Comment by wknudsen** [comment]
This seems to be not, at least entirely, a Mac bug. I am using Mac and have tested the following in both Chrome and Edge. I have both A1111 version 1.3.2 and 1.4.1 running, on both I have the latest Canvas Zoom commit cb7a3d5. On 1.4.1 built-in Canvas Zoom is disabled. 

On 1.3.2 almost everything works as expected. Hotkey Q is showing the color choice, T for pinning works. The color choice button appears on the canvas and works when clicked. Only hotkey A is not working - the dropper button is marked/highlighted when A is pressed but the thickness chooser does not appear as it does when the button is clicked. The rest of the functionality seems to work although X is kind of weird. Zooms and moves etc. with mouse scroll wheel work.

On 1.4.1 the Information hover shows, that it is the right Canvas Zoom. The color choice button does not appear. Hotkeys Q, A and T have no function. S, R work, X works in the same weird way. Zooms and moves etc. with mouse scroll wheel work.

In conclusion, the same build of Canvas Zoom works almost completely in Mac browsers on 1.3.2 but is rather limited on 1.4.1. One difference is, of course, that 1.4.1 has the builtin Canvas Zoom, although disabled.

I hope that this helps you solving the issue.

**Comment by daswer123** [comment]
Interesting of course, 1.3.2 and 1.4.1 are technically not much different, but it gives this result.

Thanks for doing the research

**Comment by wknudsen** [comment]
Here's a follow-up: I just did a clean install of 1.4.1 (through TheLastBen notebook), installed Canvas Zoom, disabled the builtin, restarted, and guess what: Canvas Zoom works, including hotkeys and color popup. I will try to install other extensions I use one by one and see if one of them breaks Canvas Zoom. I will report the results here.

**Comment by wknudsen** [comment]
I installed, one by one, some of my favourite extensions: Roop-GE, Image Browser, Regional Prompter, PhotoPea, Aspect Ratio Helper, and Dynamic Prompts. Between each install I tested that Canvas Zoom still worked. I installed After Detailer which broke the entire A1111, but it did the same in a clean install, so unrelated to Canvas Zoom.

Conclusion: Canvas Zoom basically works in Mac, any problems are probably caused by some interaction with some extension(s), just not the ones mentioned above.

**Comment by daswer123** [comment]
All right, cool. Thanks for doing this kind of research, I don't have a Mac so no way to check, really appreciate your research. 

Now there is clarity on this issue.

**Comment by wknudsen** [comment]
From a 1.4.1 installation on another server I can confirm that Canvas Zoom has no conflict with After Detailer.

**Comment by alexbabashov** [comment]
https://github.com/richrobber2/canvas-zoom/commit/398c8cb04c77cf279cdab4b94b5910cdebb1eb77
Windows 10, Google Chrome 115.0.5790.102,
1.Go to the Sketsh window
2.Upload any picture to the window
3.Press the S key for Fullscreen mode
4.Press the T key for Open color panel right above the mouse.

Expectations: 
Open color panel right above the mouse

reality: 
Open color panel above or below the cursor.
If close the window and do not change the position of the cursor and press the T key, then it works properly. After that we change the cursor position and again the window opens in the wrong place

**Comment by daswer123** [comment]
Hi, in general it should be under the cursor, it was different before , forgot to update the description. 

About the problem, found a problem, try to fix it now. Thanks for letting me know.

**Comment by daswer123** [comment]
In general, the search for the problem took more time, so only tomorrow after work I will be able to do fix.

**Comment by daswer123** [comment]
I think the Mac issue is solved, thanks @wknudsen for the research, thanks to him everything is clear now.
